### PR TITLE
Soft failures

### PR DESCRIPTION
**Pull request by imdfl**
Handle captcha, fetch and sendgrid errors

### CODE REVIEW

File: src/components/form/captcha.st.css [hunk 1]
```diff
@@ -13,6 +13,10 @@
 	animation: flickerAnimation 1s infinite;
 }
 
+.error {
+	width: 24
```
Comment (tomerlichtash) [inline src/components/form/captcha.st.css:17]: Is this necessary?

**Comment by tomerlichtash** [comment]
**Nice!**

There seem to be more work to be done to complete the UX of the form (unrelated to your changes, e.g. Success/Fail messages etc.)

I think adding a `translate` function on the fail message in the Contact Form is the minimal requirement before merge (so that untranslated string won't appear in production); otherwise, let's merge it and continue the fine-tuning afterwards.

**Comment by tomerlichtash** [comment]
@imdfl I've added a unit test for the email validation method. I've also added some mocks so I can implement an E2E by the following method:

1. Instantiate a Form component with a mock API and without a captcha.
2. Trigger a submit on the form
3. wait for a mock request with the proper data.

Will be implemented, though I believe this is not a blocker for this PR.

Please approve these changes in the meanwhile.

**Comment by imdfl** [comment]
I noticed some changes of 
```//...
}
else {
// ...
```

to `} else {`

Until we have an active development community, style enforcement is secondary to each of us feeling comfortable in their own code. Breaking long lines for readability is reasonable. Parens and block styles should be off limits, they're part of a developer's mental structure of the solution.

**Comment by tomerlichtash** [comment]
> Until we have an active development community, style enforcement is secondary to each of us feeling comfortable in their own code. Breaking long lines for readability is reasonable. Parens and block styles should be off limits, they're part of a developer's mental structure of the solution.

`if (youWantAHumanAnswer) {`

@imdfl – I'm working my Prettier, and considering as _my comfort_. I do consider uniform as top value, above my own decisions for indentation and code decisions regarding code layout.

However, while editing your code in your PRs I do keep a mental note to disable Prettier, as I'm aware of your approach towards its formatting output. However, when working on a joint PR I sometimes find it difficult to focus on enabling/disabling this feature, rather than contributing code in a comfortable environment.

Also, as per earlier agreement, from time to time we will take the code through Prettier, that's why my feeling towards this change was that is it minor.

Having said that, I will totally understand if you want to revert this change before merging this PR into the main code base. I also encourage you to come up with a solution other than Prettier, to keep our code base clean and uniform, one that you will be comfortable with.

`} else {`

Here's ChatGPT's answer to your comment:

Dear team member,

Thank you for bringing your concerns about code standards to our attention. It is important for our team to have consistent coding standards in order to ensure that our code is easy to read and maintain. However, it is also important for us to be open to different coding styles and approaches.

One possible solution to this issue is to have a team discussion about our code standards and come to a consensus on what they should be. This can include discussing which style guides or best practices we should follow, and how we can incorporate different styles and approaches in a way that is consistent and easy to understand.

Another solution is to provide guidance and training on our code standards to team members who may be new to them. This can help ensure that everyone is on the same page and can contribute to the project in a way that is consistent with our standards.

Overall, it is important for us to have consistent code standards, but it is also important to be open to different styles and approaches. By having a team discussion and providing guidance and training, we can ensure that our code is easy to read and maintain, while also incorporating different perspectives and ideas.

Best regards,
[Your Name]

`}`

**Review by imdfl — COMMENTED** [review]
No need to revert anything, it's no big deal. It's just that as long as I'm the one most likely to return to my code, it should be familiar to me at first sight. It's not a matter of principle, so changes like breaks in long param lists are reasonable, but block formatting isn't.

For future contributors, I'd rather not set prettier as a barrier to pull requests. I find this intrusive and annoying. Lint rules are fine, beyond that apply your own conventions _after_ the merge.

File: src/config/public-api-params.ts [hunk 2]
```diff
@@ -1,4 +1,10 @@
-export const SUBMIT_API = {
+export interface ISendgridSubmitAPI {
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply src/config/public-api-params.ts]: Thanks. Fixed.

**Review by tomerlichtash — APPROVED** [review]

**Pull request merged**